subsys/spm/secure_services.c

```diff
@@ -47,6 +53,42 @@ int spm_secure_services_init(void)
 }
 
 
+#ifdef CONFIG_SPM_SERVICE_READ
+typedef struct {
+	u32_t start;
+	size_t size;
+} read_range_t;
+
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_read(void *destination, const void *source, size_t num)
```

Comment:
Why is `source` `const void *` ? It seems you are mimicking `memcpy` but I see this API being called to read from an address (integer constant) rather than from pointer.

Not only you cast `source` to an integer for doing arithmetic later in this function, but also cast the address from which you read to `void *` when you are calling the function from the sample. Perhaps this function would be better with this signature: `int spm_request_read(void *destination, u32_t addr, size_t num)` to avoid these casts.

Reply:
You are right in your assumption, and I agree with your proposal, should be fixed now, please check.